File: TypingSpeedTest.js
const prompt = require('prompt-sync')();



const sentence = "The quick brown fox jumps over the lazy dog.";
console.log("Type this sentence:");
console.log(sentence);
prompt("Press Enter when ready...");

const startTime = Date.now();
const userInput = prompt("\nStart typing here:\n");
const endTime = Date.now();

// Time taken
const timeTaken = (endTime - startTime) / 1000;
const timeInMinutes = timeTaken / 60;

// Words per minute
const wordCount = sentence.split(" ").length;
const wpm = wordCount / timeInMinutes;

// Accuracy
let correctChars = 0;
for (let index = 0; index < Math.min(userInput.length, sentence.length); index++) {
if (userInput[index] === sentence[index]) {
        correctChars++;
}
}
const accuracy = (correctChars / sentence.length) * 100;

// Results
console.log("\n--- Results ---");
console.log(`Time taken: ${timeTaken.toFixed(2)} seconds`);
console.log(`Words per minute:" ${wpm.toFixed(2)}`);
console.log(`Accuracy: ${accuracy.toFixed(2)}%`);

File: TypingSpeedTest.java
import java.util.Scanner;

public class TypingSpeedTest {
public static void main(String[] args) {
String sentence = "The quick brown fox jumps over the lazy dog.";
Scanner scanner = new Scanner(System.in);

System.out.println("Type this sentence:");
System.out.println(sentence);
System.out.println("Press Enter when ready...");
        scanner.nextLine();

long startTime = System.currentTimeMillis();
System.out.println("\nStart typing here:");
String userInput = scanner.nextLine();
long endTime = System.currentTimeMillis();

        // Time taken
double timeTaken = (endTime - startTime) / 1000.0;
double timeInMinutes = timeTaken / 60.0;

        // Words per minute
int wordCount = sentence.split(" ").length;
double wpm = wordCount / timeInMinutes;

        // Accuracy
int correctChars = 0;
for (int index = 0; index < Math.min(userInput.length(), sentence.length()); index++) {
if (userInput.charAt(index) == sentence.charAt(index)) {
                correctChars++;
}
}
        double accuracy = ((double) correctChars / sentence.length()) * 100;

        // Results
System.out.printf("\n--- Results ---\n");
System.out.printf("Time taken: %.2f seconds\n", timeTaken);
System.out.printf("Words per minute: %.2f\n", wpm);
System.out.printf("Accuracy: %.2f%%\n", accuracy);

       
}
}